### ISSUE TITLE
Order of children elements under tei:person and tei:personGrp

### ISSUE THREAD

**Issue by dlschwartz** [srophe/srophe-app-data#1143]
I propose the following order (and requirements regarding frequency):

- `<elementRef key="persName" minOccurs="1" maxOccurs="unbounded"/>`
- `<elementRef key="link" minOccurs="0" maxOccurs="unbounded"/>`
- `<elementRef key="note" minOccurs="1" maxOccurs="unbounded"/>`
- `<elementRef key="birth" minOccurs="0" maxOccurs="unbounded"/>`
- `<elementRef key="death" minOccurs="0" maxOccurs="unbounded"/>`
- `<elementRef key="floruit" minOccurs="0" maxOccurs="unbounded"/>`
- `<elementRef key="gender" minOccurs="0" maxOccurs="unbounded"/>`
- `<elementRef key="state" minOccurs="0" maxOccurs="unbounded"/>`
- `<elementRef key="trait" minOccurs="0" maxOccurs="unbounded"/>`
- `<elementRef key="event" minOccurs="0" maxOccurs="unbounded"/>`
- `<elementRef key="idno" minOccurs="1" maxOccurs="unbounded"/>`
- `<elementRef key="bibl" minOccurs="1" maxOccurs="unbounded"/>`

Notes: 

- tei:link is only used to deprecate names, hence it's placement just after tei:persName
- tei:note may have @type="abstract" which suggests placement near the top but also other types: "description", "deprecation", "incerta", "misc".

**Comment by dlschwartz** [comment]
We decided to move idno up below link and floruit above birth.

- `<elementRef key="persName" minOccurs="1" maxOccurs="unbounded"/>`
- `<elementRef key="link" minOccurs="0" maxOccurs="unbounded"/>`
- `<elementRef key="idno" minOccurs="1" maxOccurs="unbounded"/>`
- `<elementRef key="note" minOccurs="1" maxOccurs="unbounded"/>`
- `<elementRef key="floruit" minOccurs="0" maxOccurs="unbounded"/>`
- `<elementRef key="birth" minOccurs="0" maxOccurs="unbounded"/>`
- `<elementRef key="death" minOccurs="0" maxOccurs="unbounded"/>`
- `<elementRef key="gender" minOccurs="0" maxOccurs="unbounded"/>`
- `<elementRef key="state" minOccurs="0" maxOccurs="unbounded"/>`
- `<elementRef key="trait" minOccurs="0" maxOccurs="unbounded"/>`
- `<elementRef key="event" minOccurs="0" maxOccurs="unbounded"/>`
- `<elementRef key="bibl" minOccurs="1" maxOccurs="unbounded"/>`

**Comment by dlschwartz** [comment]
@wlpotter Will can you please update the order of the children of tei:person and tei:personGrp to that listed in the comment above?

You can use the [schemaWork](https://github.com/srophe/srophe-app-data/tree/2023.02.21_schemaWork) branch for that.

Thanks!

**Comment by wlpotter** [comment]
Hi @dlschwartz thanks for this! I have the script running and it appears to be working. Just wanted to confirm first, though, that we don't have any cases of a record with more than one tei:person or tei:personGrp, or with both a tei:person and tei:personGrp, within the record's tei:listPerson? In other words, it's one person/personGrp per record, right?

**Comment by dlschwartz** [comment]
@wlpotter that is correct. Every record should have either one person element or one personGrp element and never both. Thanks!

**Comment by wlpotter** [comment]
Sounds good, thanks!

I also created #1145 for the documentation side of this issue, so merging the pr should be fine to close this issue

**Comment by dlschwartz** [comment]
@wlpotter Well, I counted the element names between the two branches. It looks like there are 2 fewer element names (so one element since it counts both open and close tags separately). This could mean that one element fell out but that's not the only explanation. I might poke around a bit more before I merge the pull request.

**Comment by wlpotter** [comment]
Hmm it doesn't look like we missed an element from the schema list as this is what I get when searching distinct names of children on the schemaWork branch:

persName
idno
state
floruit
bibl
note
gender
birth
death
event
link
trait

----

Compare with the branch to merge in:

persName
idno
floruit
state
bibl
note
gender
birth
death
event
link
trait

It sounds like something odd happened only with one or two records, then, right?

**Comment by dlschwartz** [comment]
I hope so. I just want to make sure. I'm working on it.

**Comment by dlschwartz** [comment]
@wlpotter I'm still not sure why I get that odd difference. It was the first thing I looked at. I used the "find in files" feature in oXygen. I restricted the xpath to "text" and selected the XML search option "element names". Then I searched for the RegEx ".+". This returns a value for the opening and the closing tag. I got two more hits on schemaWork branch than on the issue 1143 branch.

Subsequently, I used the xpath builder to search "//text//listPerson/descendant::[each descendant element]." When I do this, I get the same number of hits for each of the 33 descendant elements.

I think we can call this good as I trust the individual paths over the find in file search. I wish I had an explanation though.

I also realized that we lost 26 of the 36 comments in the original schemaWork branch. These were comments that are children of person or personGrp. The comments inside our child elements (i.e. inside note, event, etc.) were carried over in the transform. Almost all of the deleted comments were marked for deletion anyway. We only still needed one of those comments and I turned it into a GitHub issue for Dave.

I'm going to merge the branch. Thanks Will!
